src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/ApplicationInsightsLogger.cs

```diff
@@ -526,6 +526,11 @@ private void StartTelemetryIfFunctionInvocation(IDictionary<string, object> stat
                     // We'll need to store this operation context so we can stop it when the function completes
                     stateValues[OperationContext] = operation;
                 }
+            } // Track Activity if the key is the TrackActivity Scope is specified. 
+            else if (allScopes.ContainsKey("MS_TrackActivity"))
```

Comment:
Should you also check that `currentActivity != null` here? I think it'd be more clear what is happening.

Also change the comment to be:

```
// If there is a current activity, it is assumed that Application Insights will track it so we do not start an operation. 
// However, in some cases (such as Durable functions), this is not the case. This allows the scope to decide whether
// an operation should be started, even when the current activity is not null. 
```

Reply:
Hi @brettsam ,
I update the comment!  I also tried to `currentActivity != null`, however, VS19 complaint it, since the first if clause is 

```csharp
            var currentActivity = Activity.Current;
            if (currentActivity == null ||
                // Activity is tracked, but Functions wants to ignore it:
                allScopes.ContainsKey("MS_IgnoreActivity") ||
                // Functions create another RequestTrackingTelemetryModule to make sure first request is tracked (as ASP.NET Core starts before web jobs)
                // however at this point we may discover that RequestTrackingTelemetryModule is disabled by customer and even though Activity exists, request won't be tracked
                // So, if we've got AspNetCore Activity and EnableHttpTriggerExtendedInfoCollection is false - track request here.
                (!_loggerOptions.HttpAutoCollectionOptions.EnableHttpTriggerExtendedInfoCollection && IsHttpRequestActivity(currentActivity)))
            {
```

Reply:
Oh interesting -- that's okay to leave it off, I was just trying to make it more explicit. I guess it must not be null if the first check failed.